### PR TITLE
[#4649] Fix leaks observed on rule invocation (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
CI tests to be run

### CODE REVIEW

**Comment by trel** [comment]
four codacies, all the same...

**Comment by alanking** [comment]
Codacy doesn't understand how `irods::at_scope_exit` is used, so I think it is okay to ignore these.

**Comment by trel** [comment]
sold - please # when happy.

File: server/re/src/reDataObjOpr.cpp [hunk 1]
```diff
@@ -1121,15 +1151,12 @@ msiDataObjCopy( msParam_t *inpParam1, msParam_t *inpParam2,
         return rei->status;
     }
 
+    transferStat_t* transStat = NULL;
```
Comment (trel) [inline server/re/src/reDataObjOpr.cpp:1154]: nullptr?

File: server/re/src/reDataObjOpr.cpp [hunk 2]
```diff
@@ -1591,8 +1624,6 @@ int
 msiDataObjPhymv( msParam_t *inpParam1, msParam_t *inpParam2,
                  msParam_t *inpParam3, msParam_t *inpParam4, msParam_t *inpParam5,
                  msParam_t *outParam, ruleExecInfo_t *rei ) {
-    rsComm_t *rsComm;
-    dataObjInp_t dataObjInp, *myDataObjInp = NULL;
     transferStat_t *transStat = NULL;
```
Comment (trel) [inline server/re/src/reDataObjOpr.cpp:1627]: nullptr?

File: server/re/src/reDataObjOpr.cpp [hunk 3]
```diff
@@ -1704,8 +1739,6 @@ msiDataObjPhymv( msParam_t *inpParam1, msParam_t *inpParam2,
 int
 msiDataObjRename( msParam_t *inpParam1, msParam_t *inpParam2,
                   msParam_t *inpParam3, msParam_t *outParam, ruleExecInfo_t *rei ) {
-    rsComm_t *rsComm;
-    dataObjCopyInp_t dataObjRenameInp, *myDataObjRenameInp;
     dataObjInp_t *myDataObjInp = NULL;
```
Comment (trel) [inline server/re/src/reDataObjOpr.cpp:1742]: nullptr?

File: server/re/src/reDataObjOpr.cpp [hunk 4]
```diff
@@ -2299,8 +2356,6 @@ int
 msiDataObjRsync( msParam_t *inpParam1, msParam_t *inpParam2,
                  msParam_t *inpParam3, msParam_t *inpParam4, msParam_t *outParam,
                  ruleExecInfo_t *rei ) {
-    rsComm_t *rsComm;
-    dataObjInp_t dataObjInp, *myDataObjInp = NULL;
     msParamArray_t *outParamArray = NULL;
```
Comment (trel) [inline server/re/src/reDataObjOpr.cpp:2359]: nullptr?

File: server/re/src/reDataObjOpr.cpp (hunk 1 above)
Comment (alanking) [reply server/re/src/reDataObjOpr.cpp:1154]: Done

File: server/re/src/reDataObjOpr.cpp (hunk 2 above)
Comment (alanking) [reply server/re/src/reDataObjOpr.cpp:1627]: Done

File: server/re/src/reDataObjOpr.cpp (hunk 3 above)
Comment (alanking) [reply server/re/src/reDataObjOpr.cpp:1742]: Done

File: server/re/src/reDataObjOpr.cpp (hunk 4 above)
Comment (alanking) [reply server/re/src/reDataObjOpr.cpp:2359]: Done

**Comment by alanking** [comment]
CI tests passed for core, but running PREP suite to make sure.

File: server/re/src/genQueryMS.cpp [hunk 5]
```diff
@@ -347,7 +347,6 @@ msiGetMoreRows( msParam_t *genQueryInp_msp, msParam_t *genQueryOut_msp, msParam_
 
     if ( rei->status == 0 || rei->status == CAT_NO_ROWS_FOUND) {
         /* return continuation index separately in case it is needed in conditional expressions */
-        resetMsParam( continueInx );
```
Comment (alanking) [inline server/re/src/genQueryMS.cpp:350]: Looking into this as the `inpOutBuf` is being free'd here as well. If the pointers are set to `nullptr` after free'ing, the issues I found should be prevented as well as the leaks

Comment (trel) [reply server/re/src/genQueryMS.cpp:350]: that sounds correct.   can't leak a nullptr, so I've heard.

**Comment by alanking** [comment]
I think this is good to go. Still keeping an eye on PREP tests but the issue I had observed is resolved from bench testing.

**Comment by trel** [comment]
awesome, please squash to taste

**Comment by alanking** [comment]
#'d

**Pull request merged**